nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -117,6 +129,86 @@ public static DocumentModel retrieveContentFromColdStorage(CoreSession session,
+    // retrieved numberOfDaysOfAvailability in cold storage schema
+    public static ColdStorageContentStatus checkColdStorageContentAvailability(CoreSession session) {
+        log.debug("Start checking the available cold storage content for repository: {}", session::getRepositoryName);
+
+        // as the volume of result will be small, we don't use BAF
+        DocumentModelList documents = session.query(GET_DOCUMENTS_TO_CHECK_QUERY);
+
+        // for every available content we will fire an event
+        int beingRetrieved = documents.size();
+        int available = 0;
+        EventService eventService = Framework.getService(EventService.class);
+        for (DocumentModel doc : documents) {
+            if (isColdStorageContentAvailable(doc)) {
+                available++;
+                beingRetrieved--;
+                DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+                Event event = ctx.newEvent(COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME);
+                eventService.fireEvent(event);
+            }
+        }
+
+        log.debug(
+                "End checking the available cold storage content for repository: {}, beingRetrieved: {}, available: {}",
+                session.getRepositoryName(), beingRetrieved, available);
+
+        return new ColdStorageContentStatus(session.getRepositoryName(), beingRetrieved, available);
+    }
+
+    /**
+     * Checks if the retrieved cold storage content associated with the given document is available for download.
+     * 
+     * @see #retrieveContentFromColdStorage(CoreSession, DocumentRef, int)
+     * @see #checkColdStorageContentAvailability(CoreSession)
+     */
+    public static boolean isColdStorageContentAvailable(DocumentModel documentModel) {
+        // FIXME to be reworked depending on how we will check the availability on low level
+        return documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY) != null
+                && documentModel.hasFacet(FacetNames.COLD_STORAGE);
+    }
+
+    /**
+     * Status about the cold storage content being retrieved or available for a given repository.
+     */
+    public static class ColdStorageContentStatus {
+
+        protected final String repositoryName;
+
+        protected final int totalBeingRetrieved;
+
+        protected final int totalContentAvailable;
+
+        public ColdStorageContentStatus(final String repositoryName, final int totalBeingRetrieved,
+                final int totalContentAvailable) {
+            this.repositoryName = repositoryName;
+            this.totalBeingRetrieved = totalBeingRetrieved;
+            this.totalContentAvailable = totalContentAvailable;
+        }
+
+        public String getRepositoryName() {
+            return repositoryName;
+        }
+
+        public int getTotalBeingRetrieved() {
+            return totalBeingRetrieved;
+        }
+
+        public int getTotalContentAvailable() {
```

Comment:
Why is there `content` in the name of this method? It's not in the other method, and I don't think it brings anything.